web-api/src/storage.rs

```diff
@@ -47,20 +47,48 @@ use crate::{
     Error,
 };
 
-pub(crate) struct KnnSearchParams<'a, I> {
-    pub(crate) excluded: I,
+pub(crate) struct KnnSearchParams<'a> {
+    pub(crate) excluded: &'a [DocumentId],
```

Comment:
❓ why make it less flexible? for example now we can't use iterators on hashsets anymore without reallocation.

Reply:
See other comment.